store/tikv/txn.go

```diff
@@ -428,6 +431,20 @@ func (txn *tikvTxn) LockKeys(ctx context.Context, lockCtx *kv.LockCtx, keysInput
 	return nil
 }
 
+// deduplicateKeys deduplicate the keys, it use sort instead of map to avoid memory allocation.
+func deduplicateKeys(keys [][]byte) [][]byte {
+	sort.Slice(keys, func(i, j int) bool {
+		return bytes.Compare(keys[i], keys[j]) < 0
+	})
+	deduped := keys[:1]
+	for i := 1; i < len(keys); i++ {
+		if !bytes.Equal(deduped[len(deduped)-1], keys[i]) {
```

Comment:
Should we add warn logs here to remind there could be possible bugs?

Reply:
Maybe we need to support lock duplicated keys in the future.